yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -395,6 +395,10 @@ private[yarn] class YarnAllocator(
       val executorId = executorIdCounter.toString
 
       assert(container.getResource.getMemory >= resource.getMemory)
```

Comment:
memory should never be less then requested.  vcores support was added later though and if its configured off or the scheduler doesn't support it then its possible to get back less.   Like mentioned the defaultResourceCalculator just always returns 1.

There is already a comment on this at matchContainerToRequest.  Is this actually failing or you were just surprised at what you got?